llm/bill_on_document_created.py

```diff
@@ -0,0 +1,329 @@
+from firebase_functions.firestore_fn import (
+    Event,
+    DocumentSnapshot,
+)
+from llm_functions import get_summary_api_function, get_tags_api_function_v2
+from typing import TypedDict, NewType
+from collections import deque
+
+Category = NewType("Category", str)
+
+
+# This allows us to type the return of `get_topics`
+class TopicAndCategory(TypedDict):
+    # We use the name `tag` in Python, but `topic` in the database
+    topic: str
+    # Topic can be mapped directly to a category
+    category: Category
+
+
+# Using a 'deque' because appending to lists in Python is slow
+def get_categories_from_topics(
+    topics: list[str], topic_to_category: dict[str, Category]
+) -> deque[TopicAndCategory]:
+    to_return: deque[TopicAndCategory] = deque()
+    for topic in topics:
+        if topic_to_category.get(topic):
+            to_return.append(
+                TopicAndCategory(topic=topic, category=topic_to_category[topic])
+            )
+    return to_return
+
+
+# When a bill is created for a given session, we want to populate both the
+# summary and the tags for that bill. This is an idempotent function.
+def run_trigger(event: Event[DocumentSnapshot | None]) -> None:
+    bill_id = event.params["bill_id"]
+    inserted_data = event.data
+    if inserted_data is None:
```

Comment:
I think we should always expect bill creation events to have data, content, `DocumentText`, and `Title` - if they don't have these, do we need to raise an Error so that gets propagated to Google Cloud Error Reporting?